### ISSUE TITLE
Question: Is it possible to know the dynamic panel number in an html question in the panel?

### ISSUE THREAD

**Issue by Lionqueen94** [surveyjs/survey-library#2583]
### Are you requesting a feature, reporting a bug or asking a question?
Asking a question

### What is the current behavior?
I have a dynamic panel with some textfields. Once all textfields are filled, a button appears (html question) that calls a javascript function onclick. This Javascript function is meant to call an api with the data filled in in that panel. I do have access to the data in Javascript, through survey.data, but since it's a dynamic panel, the input in the text fields is stored in an array. I need to know from which panel the button was clicked, to know which item in the array I need to use. Is there a way to get the panel number in an html question?

Sidenote, what I'm currently doing is sending the input in the text fields to the backend (with `onclick="javascriptFunction({\"fieldname": \"{panel.fieldname}\", ...})"`), this is however subjective to XSS and I need to find a way to prevent this.

### Specify your
* browser: Chrome/Firefox
* browser version: latest
* surveyjs platform (angular or react or jquery or knockout or vue): jquery
* surveyjs version: 1.8.21

**Comment by andrewtelnov** [comment]
@Lionqueen94 There is a [parent](https://surveyjs.io/Documentation/Library?id=questiontextmodel#parent) and [parentQuestion](https://surveyjs.io/Documentation/Library?id=questiontextmodel#parentQuestion) properties in the question object.
In your case `parentQuestion` is not null, it is a panel dynamic and `parent` is the panel where question is located.
To get the panel index you have to use the folllowing code: `question.parentQuestion.panels.indexOf(question.panel)`.
Regarding your html question. I would stronly recommend you to create a custom widget and use it. Here is [the docs](https://surveyjs.io/Documentation/Survey-Creator?id=Create-Custom-Widget).

Thank you,
Andrew

**Comment by Lionqueen94** [comment]
@andrewtelnov Thanks for your fast response. I have 2 follow up questions.
1. If I want to use a widget, would I then link it to one of my 4 text fields and then check if the other 3 text fields are also not empty before showing the widget with the buttons? Or can I link it to the panel and show the widget when the text fields in the panel are notempty?
2. Is there a way to send the current question object when calling the Javascript function in my html question?

**Comment by andrewtelnov** [comment]
@Lionqueen94 I would leve your current questions as it is and make a question that will be actually a button. You will have access to the question object in rendering function of your custom widget. Please review [this example](https://plnkr.co/edit/HdnYE5?p=preview&preview).

Thank you,
Andrew

**Comment by Lionqueen94** [comment]
@andrewtelnov Thanks for this example! I was able to get it working with the widget and do not have to worry about XSS anymore. The only problem I still have is getting the panel index. `question.panel` is undefined in the widgets `afterRender` function which makes the `question.parentQuestion.panels.indexOf(question.panel)` return -1 at all times.

**Comment by andrewtelnov** [comment]
@Lionqueen94 Yes, right. I did not use the code correctly, it should be `question.parent` of course:
 `question.parentQuestion.panels.indexOf(question.parent)`

Thank you,
Andrew

**Comment by Lionqueen94** [comment]
@andrewtelnov Thanks a lot, it's fully working now, amazing! :+1:
